devbuild.sh

```diff
@@ -1,76 +1,235 @@
 #!/bin/bash
-set -eu
-
-#cd to location of script
-MYDIR=$(cd "$(dirname "$(readlink -f -n "${BASH_SOURCE[0]}" )" )" && pwd -P)
 
+# usage instructions
 usage () {
-  echo "Usage: "
-  echo "  $0 PLATFORM COMPILER"
-  echo ""
-  echo "PLATFORM: Name of machine you are building on"
-  echo "COMPILER: (optional) compiler to use; valid options are 'intel', 'gnu'"
-  echo ""
-  echo "NOTE: This script is for internal developer use only;"
-  echo "See User's Guide for detailed build instructions"
+  printf "Usage: $0 [OPTIONS]...\n"
+  printf "\n"
+  printf "OPTIONS\n"
+  printf "  -h, --help\n"
+  printf "      show this help guide\n"
+  printf "  --platform=PLATFORM\n"
+  printf "      name of machine you are building on; defauts to hostname\n"
+  printf "      (.e.g. cheyenne | hera | jet | orion | wcoss)\n"
+  printf "  --compiler=COMPILER\n"
+  printf "      compiler to use; default depends on platform\n"
+  printf "      (.e.g. intel | gnu | cray | gccgfortran)\n"
+  printf "  --app=APPLICATION\n"
+  printf "      weather model application to build\n"
+  printf "      (.e.g. ATM | ATMW | S2S | S2SW)\n"
+  printf "  --ccpp=\"CCPP_SUITE1,CCPP_SUITE2...\"\n"
+  printf "      CCCP suites to include in build; delimited with ','\n"
+  printf "  --components=\"COMPONENT1,COMPONENT2...\"\n"
```

Comment:
What are components in this context?  Like the atmosphere component, ocean component, etc?  And what are the valid values that can be included in the list?

Reply:
@gsketefian This feature is actually the reason I modified the devbuild script. I found this helpful because this feature allows me to add components to the build without modifying the build script or any CMake files.  I will be adding a "WRFHYDRO" component for coupling RRFS-NWM. I tested this and it doesn't conflict with --app, so you can say --app=ATM --components=WW3 and it will build the standard ATM system and add WW3. I can add an example list here ("e.g. "). The link below gives you an example of the available build components.

[https://github.com/ufs-community/ufs-weather-model/blob/develop/CMakeLists.txt#L25](https://github.com/ufs-community/ufs-weather-model/blob/develop/CMakeLists.txt#L25)

Reply:
Yes, it would be great if you add some examples.  In addition, it would be good if you can mention the location of that CMakeLists.txt file so people can see what the latest components are.  Thx.

Reply:
Added examples and note about CMakeLIsts.txt

Reply:
That's great, thanks!